src/hooks/useAdTimeDetector.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useRef } from "react";
+
+const isAdTimeNow = (minutes, seconds) => {
+  const isPreThirtyAdTime =
+    (minutes === 27 && seconds >= 30) ||
+    minutes === 28 ||
+    minutes === 29 ||
+    (minutes === 30 && seconds === 0);
+
+  const isPreHourAdTime =
+    minutes === 57 ||
+    minutes === 58 ||
+    minutes === 59 ||
+    (minutes === 0 && seconds === 0);
+
+  return isPreThirtyAdTime || isPreHourAdTime;
+};
+
+const useAdTimeDetector = (handleChannelSwitch) => {
+  const isAdTimeAlreadyHandled = useRef(false);
+
+  useEffect(() => {
+    const checkAdTime = () => {
+      const now = new Date();
+      const minutes = now.getMinutes();
+      const seconds = now.getSeconds();
+
+      const isAdTime = isAdTimeNow(minutes, seconds);
```

Comment:
아니면 `minutes`와 `seconds`는 함수 전달하는 한 번만 사용되니까
```suggestion
      const isAdTime = isAdTimeNow(now.getMinutes(), now.getSeconds());
```
이렇게 할 수도 있겠네요! @Eun0713 @faber-h 두 분은 어떻게 생각하시나요?

Reply:
현재는 한 번만 호출되기 때문에, 별도로 변수로 분리하기보다는 함수 호출 시 바로 전달하는 방식이 더 간결하고 적절할 것 같아요~

Reply:
그렇네요! 변수로 따로 빼는 것보다 바로 함수에 넘기는 게 더 간결하고 보기 좋은 것 같아요. 반영했습니다!
말씀 주셔서 감사합니다 :)